Source/Dafny/Resolver.cs

```diff
@@ -12800,6 +12800,10 @@ private class ClonerKeepLocalVariablesIfTypeNotSet : Cloner {
         // Check if the head of the pattern is a constructor or a variable
         if (ctors.TryGetValue(idpat.Id, out ctor)) {
           /* =[3]= */
+          if (ctor != null && idpat.Arguments == null && ctor.Formals.Count != 0) {
+            this.reporter.Warning(MessageSource.Resolver, pat.Tok,
+              $"Variable name '{pat}' has the same name as an exisiting constructor");
```

Comment:
`exisiting` has a typo, but let's say `a datatype constructor that's in scope at this location`